includes/event/event-capabilities.php

```diff
@@ -0,0 +1,186 @@
+<?php
+
+namespace Wporg\TranslationEvents\Event;
+
+use GP;
+use WP_User;
+use Wporg\TranslationEvents\Attendee\Attendee;
+use Wporg\TranslationEvents\Attendee\Attendee_Repository;
+use Wporg\TranslationEvents\Stats\Stats_Calculator;
+
+class Event_Capabilities {
+	private const CREATE = 'create_translation_event';
+	private const VIEW   = 'view_translation_event';
+	private const EDIT   = 'edit_translation_event';
+	private const DELETE = 'delete_translation_event';
+
+	/**
+	 * All the capabilities that concern an Event.
+	 */
+	private const CAPS = array(
+		self::CREATE,
+		self::VIEW,
+		self::EDIT,
+		self::DELETE,
+	);
+
+	private Event_Repository_Interface $event_repository;
+	private Attendee_Repository $attendee_repository;
+	private Stats_Calculator $stats_calculator;
+
+	public function __construct(
+		Event_Repository_Interface $event_repository,
+		Attendee_Repository $attendee_repository,
+		Stats_Calculator $stats_calculator
+	) {
+		$this->event_repository    = $event_repository;
+		$this->attendee_repository = $attendee_repository;
+		$this->stats_calculator    = $stats_calculator;
+	}
+
+	/**
+	 * This function is automatically called whenever user_can() is called for one the capabilities in self::CAPS.
+	 *
+	 * @param string  $cap  Requested capability.
+	 * @param array   $args Arguments that accompany the requested capability check.
+	 * @param WP_User $user User for which we're evaluating the capability.
+	 * @return bool
+	 */
+	private function has_cap( string $cap, array $args, WP_User $user ): bool {
+		switch ( $cap ) {
+			case self::CREATE:
+				return $this->has_create( $user );
+			case self::VIEW:
+			case self::EDIT:
+			case self::DELETE:
+				if ( ! isset( $args[2] ) || ! is_int( $args[2] ) ) {
+					return false;
+				}
+				$event = $this->event_repository->get_event( $args[2] );
+				if ( ! $event ) {
+					return false;
+				}
+
+				if ( self::VIEW === $cap ) {
+					return $this->has_view( $user, $event );
+				} elseif ( self::EDIT === $cap ) {
```

Comment:
Since we're returning, we can remove the `else`.
```suggestion
				}
				if ( self::EDIT === $cap ) {
```